src/util/elf2efi.c

```diff
@@ -127,6 +127,34 @@
 #define R_ARM_V4BX 40
 #endif
 
+/* LoongArch */
+#ifndef EM_LOONGARCH
+#define EM_LOONGARCH 258	/* LoongArch */
+#endif
+
+/* Seems to be missing from elf.h */
+#ifndef R_LARCH_NONE
+#define R_LARCH_NONE        0
+#endif
+#ifndef R_LARCH_64
+#define R_LARCH_64          2
+#endif
+#ifndef R_LARCH_B26
+#define R_LARCH_B26         66
```

Comment:
This (and the following definitions) seem not to be present in even the upstream master glibc `elf.h`: https://sourceware.org/git/?p=glibc.git;a=blob;f=elf/elf.h;h=b6a75c13a84bb04aff9c70fdfa63a1a681f29b33;hb=refs/heads/master#l4102

Where are these definitions coming from?

Reply:
Thank you for the reference.

The whole "constants missing on some platforms" section of `elf2efi.c` is just a list of definitions that do exist in glibc's `<elf.h>` but may be missing on the user's build host since the user may have a slightly older version of `<elf.h>` currently installed.  I'd prefer to keep it this way.

Is there a reason why these constants are not present in upstream glibc?  It seems as though they should also be present there.

Reply:
Yes, you're right, it should exist in `<elf.h>`, but it's not there yet, so in ipxe I added the define, and the next glibc version should have it.

Reply:
Thanks. Let's get those definitions merged into glibc first - what's the expected timeline for that?

Reply:
Not soon, maybe in glibc 2.38.
https://patchwork.sourceware.org/project/glibc/patch/20230131133246.711897-1-caiyinyu@loongson.cn/

Reply:
Understood.  As soon as they get merged to the development (master?) branch in glibc, then I'm happy to take those conditional `#define`s into iPXE.

Reply:
The patch was merged.
https://sourceware.org/git/?p=glibc.git;a=commit;h=83d49a53aab1c8e191c007bc986c4d696bc95570